docs: Add Architecture Overview Assets (#1780)
hiero-ledger/hiero-block-node: docs/block-node/architecture/plugins.md

# BlockNode Plugin System & Extensibility

The BlockNode is designed for extensibility through a robust plugin system.
All major features are implemented as plugins, allowing developers to add, modify, or replace functionality easily.

## Plugin Architecture

- **Base Interface:** All plugins implement the `BlockNodePlugin` interface, which defines lifecycle methods
  (`init`, `start`, `stop`) and configuration hooks.
- **Messaging Facility:** The `BlockMessagingFacility` is a special plugin responsible for distributing events
  (block items or notifications) to registered handlers (other plugins/components). This is how plugins communicate and
  coordinate processing logic between themselves.
- **Service Loading:** Plugins are discovered and loaded dynamically at startup using a service loader pattern (`ServiceLoaderFunction`).

## Plugin Lifecycle

1. **Discovery:** On startup, the `BlockNodeApp` scans for available plugins implementing `BlockNodePlugin` using JPMS ServiceLoader.
2. **Initialization:** Each plugin is initialized with the application context and appropriate service routing builders.

> 3. **Start:** Plugins are started, enabling them to process events and interact with other components.

Reviewer: "register to receive and process events" ?

4. **Stop:** On shutdown, plugins are stopped gracefully.

## Adding a New Plugin

To add a new plugin:

1. Implement the `BlockNodePlugin` interface in your module.

> 2. Optionally, extend specialized interfaces (e.g., `BlockProviderPlugin`, `HistoricalBlockFacility` or `BlockMessagingFacility`) for additional around block management and messaging capabilities.

Reviewer 2: These mostly _replace_ for messaging or extend (for provider or historical).
Also, interfaces are implemented, not extended.
```suggestion
2. Optionally, implement specialized interfaces (e.g., `BlockProviderPlugin`, `HistoricalBlockFacility` or `BlockMessagingFacility`) if your plugin replaces or extends block management and messaging capabilities.
```

# BlockNode Plugin System & Extensibility

The BlockNode is designed for extensibility through a robust plugin system.
All major features are implemented as plugins, allowing developers to add, modify, or replace functionality easily.

## Plugin Architecture

- **Base Interface:** All plugins implement the `BlockNodePlugin` interface, which defines lifecycle methods
  (`init`, `start`, `stop`) and configuration hooks.
- **Messaging Facility:** The `BlockMessagingFacility` is a special plugin responsible for distributing events
  (block items or notifications) to registered handlers (other plugins/components).
  - This facility is how plugins communicate data and events between plugins.
  - This facility may be replaced, but a single implementation must be present in any Block Node.

> - **Service Loading:** Plugins are discovered and loaded dynamically at startup using the JPMS a service loader.

Reviewer 2: Extra `a`.
```suggestion
- **Service Loading:** Plugins are discovered and loaded dynamically at startup using the JPMS service loader.
```
